### DISCUSSION TITLE
Application of casadi in rapid control prototype

### DISCUSSION THREAD

**Discussion by BowlingLuka** [casadi/casadi#3095] [Q&A]
Recently, I encountered some issues while testing the real-time performance of a control strategy in a rapid control prototype (RCP) experiment. I am using the CasADi toolbox to solve the optimization problem. The controller is written in the 'm.file' format in Simulink using the 'Matlab Function' block, but I encountered errors ("Import statements are currently unsupported.") during the application building process. I would like to know how to use CasADi to translate the controller from Simulink to C/C++ for code generation.

Note: It is recommended not to declare a function as extrinsic (coder.extrinsic(*)) as it may result in errors during code generation.

**Comment by jgillis** [comment]
Dear @BowlingLuka , the keyword to search for is CasADi code generation, see e.g. https://web.casadi.org/blog/s-function/
Perhaps Matlab Coder can be made to automatically perform CasADi code generation, but I have not looked into that.

**Reply by BowlingLuka** [reply]
Dear @jgillis , thank you for your reply.
Before starting the discussion, I had already read the codegen of Casadi's blog. However, I am sorry there is no way to resolve nonlinear OCPs. Can you give me a demo of a solution for nonlinear OCP (e.g. Nonlinear and quadratic programming on casadi official website Home)?
